File: android/app/src/main/kotlin/com/example/smigoal/db/MessageEntity.kt
package com.example.smigoal.db

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "MESSAGE_TABLE")
data class MessageEntity(
    @PrimaryKey(autoGenerate = true) val id: Int? = null,
    val url: List<String>?,
    val message: String,
    val sender: String,
    var thumbnail: String,
    val containsUrl: Boolean,
    val timestamp: Long,
    var hamPercentage: Double,
    var spamPercentage: Double,
    var isSmishing: Boolean
) {
    constructor(url: List<String>?, message: String, sender: String, thumbnail: String, containsUrl: Boolean, timestamp: Long, hamPercentage: Double, spamPercentage: Double, isSmishing: Boolean)
            : this(null, url, message, sender, thumbnail, containsUrl, timestamp, hamPercentage, spamPercentage, isSmishing)
}

File: lib/models/message_entity.dart
class MessageEntity {
  final int? id;
  final List<String>? url;
  final String message;
  final String sender;
  final String thumbnail;
  final bool containsUrl;
  final int timestamp;
  final bool isSmishing;

  MessageEntity({
    required this.id,
    required this.url,
    required this.message,
    required this.sender,
    required this.thumbnail,
    required this.containsUrl,
    required this.timestamp,
    required this.isSmishing
  });

  factory MessageEntity.fromMap(Map<String, dynamic> map) {
    List<String>? urls;
    int? tid;
    if (map['id'] != null) {
      tid = map['id'];
    }
    if (map['url'] != null) {
      urls = List<String>.from(map['url']);
    }
    return MessageEntity(
      id: tid,
      url: urls,
      message: map['message'] as String,
      sender: map['sender'] as String,
      thumbnail: map['thumbnail'] as String,
      containsUrl: map['containsUrl'] as bool,
      timestamp: map['timestamp'] as int,
      isSmishing: map['isSmishing'] as bool
    );
  }
}